.github/workflows/ci.yml

```diff
@@ -78,9 +75,11 @@ jobs:
         run: Scripts/github/prepare-simulators.sh ${{ matrix.platforms }}
       - name: Build and Test Framework
         run: Scripts/build.swift ${{ matrix.platforms }}
+      - name: Prepare Coverage Reports
+        run: ./Scripts/prepare-coverage-reports.sh
```

Comment:
Unfortunately, `codecov/codecov-action@v3`'s `xcode_archive_path` parameter does not seem to work in CI (I could get it working locally but not in CI), so instead of pointing CI at our `.xcresult` files, we instead export our own `lcov` files using a script.

This isn't how I'd like to do this, but it works. And I spent a few too many hours attempting other solutions before getting this solution working. At this point: if the reviewer would like us to generate reports a different way... pull requests welcome 😬